File: 02/b.nim
import strutils, utils

var result = 0
for line in load().splitLines:
  let a = int(line[0]) - int('A')
  let x = int(line[2]) - int('X')
  result += 3 * x + 1 + (2 + x + a) mod 3
echo result

File: 02/b.cpp
#include "utils.h"

long long solve(vector<string> &lines) {
  long long result = 0;

  for (string &line : lines) {
    stringstream ss(line);
    char abc, xyz;
    ss >> abc >> xyz;
    const int a = abc - 'A', x = xyz - 'X';
    result += 3 * x + 1 + (2 + x + a) % 3;
  }

  return result;
}